app/commands/create_election.rb

```diff
@@ -47,93 +50,84 @@ def build_election
+    @invalid_message ||= log_entry_validations || election_validations || questions_validations || election.voting_scheme.validate_election
     @invalid_message.present?
   end
 
   def create_trustees
     trustees.each do |trustee|
-      create_trustee(trustee)
+      find_or_create_trustee(trustee).election_trustees.create!(election: election)
     end
   end
 
-  def create_trustee(trustee)
-    trustee_attributes = {
-      name: trustee_name(trustee),
-      public_key: trustee_public_key(trustee)
-    }
-    t = Trustee.where(name: trustee_name(trustee)).or(Trustee.where(public_key: trustee_public_key(trustee))).first
-    t = Trustee.create!(trustee_attributes) if t.blank?
-    t.elections_trustees.create!(election: election)
+  def find_or_create_trustee(trustee)
+    Trustee.where(name: trustee["name"]).or(Trustee.where(public_key: trustee["public_key"])).first ||
+      Trustee.create!(
+        name: trustee["name"],
+        public_key: trustee["public_key"]
+      )
   end
 
   def trustees
-    decoded_data.dig("trustees")
-  end
-
-  def trustee_name(trustee)
-    trustee.dig("name")
+    @trustees ||= decoded_data["trustees"]
   end
 
-  def trustee_public_key(trustee)
-    trustee.dig("public_key")
+  def election_id
+    @election_id ||= decoded_data["election_id"]
   end
 
   def title
-    decoded_data.dig("description", "name", "text", 0, "value")
+    @title ||= decoded_data.dig("description", "name", "text", 0, "value")
   end
 
   def start_date
-    start_date = decoded_data.dig("description", "start_date")
-    Time.zone.parse(start_date) if start_date.present?
+    return @start_date if defined?(@start_date)
+
+    @start_date ||= Time.zone.parse(decoded_data.dig("description", "start_date") || "")
   end
 
   def end_date
-    end_date = decoded_data.dig("description", "end_date")
-    Time.zone.parse(end_date) if end_date.present?
-  end
+    return @end_date if defined?(@end_date)
 
-  def questions
-    decoded_data.dig("description", "contests")
+    @end_date ||= Time.zone.parse(decoded_data.dig("description", "end_date") || "")
   end
 
-  def answers_validations
-    questions.each do |quest|
+  def questions
+    @questions ||= decoded_data.dig("description", "contests")
+  end
+
+  def election_validations
+    @election_validations = if decoded_data.blank?
+                              "Invalid signature"
+                            elsif decode_error.present?
+                              decode_error
+                            elsif election.voting_scheme_class.blank?
+                              "A valid Voting Scheme must be specified"
+                            elsif title.blank?
+                              "Missing title"
+                            elsif start_date.after?(end_date)
+                              "Starting date cannot be after the end date"
+                            elsif start_date.before?(Time.current + 2 * 60 * 60)
```

Comment:
Can we maybe use `2.hours.from_now` instead of `(Time.current + 2 * 60 * 60)` ? I think, it's easier to understand.

Reply:
Oh, I didn't see that one! Thanks! ❤️